### PR TITLE
fix(lib/txmgr): handle wait mine edge case

### PR DESCRIPTION
**Pull request by corverroos**
Fixes root cause of "nonce to low" in e2e tests. When quering txreceipts, geth sometimes returns "transaction indexing in progress". Old txmgr didn't handle this and tried resubmitting while the tx was then confirmed in most cases. Instead it should re-attempt fetching the receipt.

task: none

### CODE REVIEW

**Review by Lazar955 — APPROVED** [review]

**Comment by corverroos** [comment]
```
24-02-23 15:04:04.631 DEBU Txmgr crafted new tx                     req_id=54e4722 chain=omni_evm nonce=0 tx=0x79bf96e17e1cc8eb5c47b6349da29e258c3bbe3c91046162a7229293c8bd8286 gas_tip_cap=1000000000 gas_fee_cap=3000000000 gas_limit=447962
24-02-23 15:04:04.970 WARN Transaction receipt not mined, probably replaced req_id=54e4722 chain=omni_evm err="transaction indexing is in progress"
24-02-23 15:04:52.638 DEBU Bumping gas price                        req_id=54e4722 chain=omni_evm
24-02-23 15:04:52.643 DEBU Updating fees                            req_id=54e4722 chain=omni_evm old_gas_tip_cap=1000000000 old_gas_fee_cap=3000000000 new_gas_tip_cap=1000000000 new_gas_fee_cap=1018756058 new_base_fee=9378029
24-02-23 15:04:52.643 DEBU Using threshold tip and threshold feecap req_id=54e4722 chain=omni_evm
24-02-23 15:04:52.648 WARN Nonce too low                            req_id=54e4722 chain=omni_evm err="nonce too low: next nonce 1, tx nonce 0"^C
```

File: lib/txmgr/txmgr.go [hunk 1]
```diff
@@ -493,7 +499,9 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txHash common.Hash,
 	defer cancel()
 	receipt, err := m.backend.TransactionReceipt(ctx, txHash)
 	if err != nil {
-		if errors.Is(err, ethereum.NotFound) || strings.Contains(err.Error(), ethereum.NotFound.Error()) {
+		if strings.Contains(err.Error(), "transaction indexing is in progress") {
```
[the remark this answers is not in the document]
Comment (pavelnikolov) [reply lib/txmgr/txmgr.go:500]: Not [a Go-idiomatic way of working with errors](https://go.dev/blog/go1.13-errors). We should always use `errors.Is` and `errors.As` to check if an error is of particular type.

[the remark this answers is not in the document]
Comment (pavelnikolov) [reply lib/txmgr/txmgr.go:500]: I just realized that this error is inside of the `internal` ethereum package and we can't call `errors.Is` on it 🤦 .
I'll raise an issue on the Ethereum repo.

[the remark this answers is not in the document]
Comment (pavelnikolov) [reply lib/txmgr/txmgr.go:500]: I submit [an issue upstream](https://github.com/ethereum/go-ethereum/issues/29063) to make that error public. The docs say that there is no receipts for pending transactions. Alternatively, we can try to get the transaction by hash and verify that it's not pending and only then get the receipt without any string comparison. Otherwise, if the error text changes, our code will stop working as expected with a silent bug.
